mgmt/rest/v2/plugin.go

```diff
@@ -227,6 +259,10 @@ func (s *apiV2) loadPlugin(w http.ResponseWriter, r *http.Request, _ httprouter.
 			return
 		}
 		rp.SetSignature(signature)
+		rp.SetCACertPaths(caCertPaths)
+		rp.SetCertPath(certPath)
+		rp.SetKeyPath(keyPath)
+		rp.SetTLSEnabled(isTLSEnabled(certPath, keyPath))
```

Comment:
I understand the goal, however, it looks like inception.  But what is more, there is no handling of error when one of pair is not provided. 
I suggest to change it as it is done in API v1 - see https://github.com/intelsdi-x/snap/blob/master/mgmt/rest/v1/plugin.go#L182
```
if certPath != "" && keyPath != "" {
	rp.SetTLSEnabled(true)
} else if certPath != "" || keyPath != "" {
	e := errors.New("Error: TLS setup incomplete - missing one of pair: certificate, key files")
	rbody.Write(500, rbody.FromError(e), w)
	return
}
```